diff --git a/dist/solid-refresh.mjs b/dist/solid-refresh.mjs
index d69390a4ee6afd5544e56722da922d6c03a15ce8..e8192141e5a80b3496b29ee88d50ff17a66cae8d 100644
--- a/dist/solid-refresh.mjs
+++ b/dist/solid-refresh.mjs
@@ -38,6 +38,9 @@ function createProxy(source, name, location) {
     }
     return new Proxy(HMRComp, {
         get(_, property) {
+            if (property === 'location' || property === 'name') {
+                return HMRComp[property];
+            }
             return source()[property];
         },
         set(_, property, value) {